[MJAVADOC-489] Find the main module descriptor

io7m opened the pull request:
The existing code was checking for the existence of a file
"src/main/module-info.java" relative to the current directory. The
problem is that in a multi-module build, the current directory is
the project root, not the root of the current module. As a result,
the module descriptor was not being found and all dependencies were
ending up on the classpath instead.

Fixes: MJAVADOC-489

io7m committed:
[MJAVADOC-489] Find the main module descriptor

The existing code was checking for the existence of a file
"src/main/module-info.java" relative to the current directory. The
problem is that in a multi-module build, the current directory is
the project root, not the root of the current module. As a result,
the module descriptor was not being found and all dependencies were
ending up on the classpath instead.

Fixes: MJAVADOC-489

eolivelli commented on maven-javadoc-plugin/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java:
```diff
@@ -4739,6 +4741,19 @@ private void addJavadocOptions( List<String> arguments, List<String> sourcePaths
+        for ( String root : roots )
+        {
+            File descriptorFile = new File( root, "module-info.java" ).getAbsoluteFile();
+            if ( descriptorFile.exists() )
```
Won't be better isFile() instead of exists() ?

io7m replied:
I agree, I'll update shortly.

io7m replied:
Actually, I'm kind of inclined to fail hard and fast if something exists called `module-info.java` that isn't a regular file. That almost certainly indicates some sort of problem in the build elsewhere.

eolivelli replied:
Totally makes sense +1

io7m committed:
Fail quickly on non-file module-info.java

This adjusts the previous patch to quickly and loudly reject filesystem
entries named module-info.java that turn out not to be regular files.

Affects: MJAVADOC-489

eolivelli reviewed (approved):
+1 non binding

io7m commented:
It looks like only half of this got merged as I didn't use the same ticket title on the second commit.
